File: tutorial/code/ts/22-symmetric.ts
import { Panproto, SymmetricLensHandle } from '@panproto/core';

const panproto = await Panproto.init();

// Two teams with different schemas — neither is "old" or "new"
const teamA = buildTeamASchema(panproto);
const teamB = buildTeamBSchema(panproto);

// Auto-generate a symmetric lens
using sync = SymmetricLensHandle.fromSchemas(teamA, teamB, panproto._wasm);

// Sync A's data to B's view
const { view: bView, complement: bComplement } = sync.syncLeftToRight(
  teamAData, teamAComplement
);

// Sync B's data to A's view
const { view: aView, complement: aComplement } = sync.syncRightToLeft(
  teamBData, teamBComplement
);

File: tutorial/code/py/22-symmetric.py
from panproto import Panproto, SymmetricLensHandle

panproto = Panproto.init()

with SymmetricLensHandle.from_schemas(team_a, team_b, panproto._wasm) as sync:
    b_view, b_comp = sync.sync_left_to_right(a_data, a_complement)
    a_view, a_comp = sync.sync_right_to_left(b_data, b_complement)
